_unittest/test_21_Circuit.py

```diff
@@ -484,3 +481,122 @@ def test_38_browse_log_file(self):
     def test_39_export_results_circuit(self):
         exported_files = self.aedtapp.export_results()
         assert len(exported_files) > 0
+
+    def test_40_assign_sources(self):
+        self.aedtapp.insert_design("sources")
+        c = self.aedtapp
+        filepath = os.path.join(local_path, "example_models", test_subfolder, "frequency_dependent_source.fds")
+        assert c.create_source(source_type="PowerSin", name="PowerSinusoidal3")
+        c.sources["PowerSinusoidal3"].ac_magnitude = "2V"
+        c.sources["PowerSinusoidal3"].ac_phase = "2deg"
+        c.sources["PowerSinusoidal3"].dc_magnitude = "2V"
+        c.sources["PowerSinusoidal3"].power_offset = "2W"
+        c.sources["PowerSinusoidal3"].power_magnitude = "20W"
+        c.sources["PowerSinusoidal3"].frequency = "20GHz"
+        c.sources["PowerSinusoidal3"].delay = "20ns"
+        c.sources["PowerSinusoidal3"].damping_factor = "200"
+        c.sources["PowerSinusoidal3"].phase_delay = "100deg"
+        c.sources["PowerSinusoidal3"].tone = "100Hz"
+
+        assert c.create_source(source_type="PowerIQ", name="PowerIQ")
+        c.sources["PowerIQ"].carrier_frequency = "2GHz"
+        c.sources["PowerIQ"].sampling_time = "2s"
+        c.sources["PowerIQ"].dc_magnitude = "2V"
+        c.sources["PowerIQ"].repeat_from = "22s"
+        c.sources["PowerIQ"].delay = "20ns"
+        c.sources["PowerIQ"].carrier_amplitude_voltage = "20V"
+        c.sources["PowerIQ"].carrier_amplitude_power = "20W"
+        c.sources["PowerIQ"].carrier_offset = "100V"
+        c.sources["PowerIQ"].real_impedance = "100ohm"
+        c.sources["PowerIQ"].imaginary_impedance = "1000ohm"
+        c.sources["PowerIQ"].damping_factor = "200"
+        c.sources["PowerIQ"].phase_delay = "100deg"
+        c.sources["PowerIQ"].tone = "100Hz"
+        c.sources["PowerIQ"].i_q_values = [["0s", "1V", "2V"], ["1s", "3V", "2V"]]
+        c.sources["PowerIQ"].file = filepath
+
+        source_freq = c.create_source(source_type="VoltageFrequencyDependent", name="freq_pyaedt")
+        source_freq.magnitude_angle = True
+        source_freq.frequencies = [2000000000, 3000000000]
+        source_freq.vmag = [0.001, 0.02]
+        source_freq.vang = [0.0349065850398866, 0.0872664625997165]
+        source_freq.vreal = [2, 1]
+        source_freq.vimag = ["0.2", "0.1"]
+        source_freq.magnitude_angle = False
+        source_freq.fds_filename = filepath
+        source_freq.fds_filename = None
+        assert source_freq.vreal
```

Comment:
maybe is good to check and assert that also the value is according to what expected